Sources/Client/Extensions/WeakRef.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  WeakRef.swift
+//  StreamChatClient
+//
+//  Created by Alexey Bukhtin on 29/01/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+final class WeakRef<T: AnyObject> {
+    weak var value: T?
+    
+    init(_ value: T) {
+        self.value = value
+    }
+}
+
+extension Array {
+    /// Remove all nil values.
+    mutating func flush<T: AnyObject>() where Element == WeakRef<T> {
+        if isEmpty {
+            return
+        }
+        
+        // Find indices with the nil value.
+        // For example values with indices: [0: nil, 1: value1, 2: nil, 3: value2]
+        let nilIndices = enumerated()
```

Comment:
Instead of all this, we can just write `self = filter { $0.value != nil }` and it works :)